### PR TITLE
aurora: unauthorized-fetch

### PR DESCRIPTION
**Pull request by domdinicola**
[AB#199333](https://unicef.visualstudio.com/4e044e8d-bc28-4768-8074-f80ff6e4a0e5/_workitems/edit/199333)

### CODE REVIEW

File: backend/hct_mis_api/aurora/admin.py [hunk 1]
```diff
@@ -205,17 +205,20 @@ def fetch_aurora(self, request: HttpRequest) -> HttpResponse:
...
-                if form.cleaned_data["registration"]:
-                    filters["registration"] = form.cleaned_data["registration"].source_id
-                info = fetch_records(aurora_token, form.cleaned_data["overwrite"], **filters)
-                ctx["info"] = info
+            try:
+                if "detect" in request.POST:
+                    ctx["metadata"] = get_metadata(aurora_token)
+                elif form.is_valid():
+                    filters = {
+                        "id": form.cleaned_data.get("from_id", "") or "",
```
Comment (KonradMarzec1991) [inline backend/hct_mis_api/aurora/admin.py:213]: This `or ""` seems to me unnecessary

File: backend/hct_mis_api/aurora/admin.py [hunk 2]
```diff
@@ -205,17 +205,20 @@ def fetch_aurora(self, request: HttpRequest) -> HttpResponse:
...
+                        "after": form.cleaned_data.get("after_date", "") or "",
+                    }
+                    if form.cleaned_data["registration"]:
```
Comment (KonradMarzec1991) [inline backend/hct_mis_api/aurora/admin.py:216]: This part will raise KeyError everytime when `registration` is not in cleaned_data

File: backend/hct_mis_api/aurora/admin.py [hunk 3]
```diff
@@ -205,17 +205,20 @@ def fetch_aurora(self, request: HttpRequest) -> HttpResponse:
...
+                        filters["registration"] = form.cleaned_data["registration"].source_id
+                    info = fetch_records(aurora_token, form.cleaned_data["overwrite"], **filters)
+                    ctx["info"] = info
+            except BaseException as e:
```
Comment (KonradMarzec1991) [inline backend/hct_mis_api/aurora/admin.py:220]: Why `BaseException`?

File: backend/hct_mis_api/aurora/admin.py (hunk 1 above)
Comment (domdinicola) [reply backend/hct_mis_api/aurora/admin.py:213]: mmm cleaned_data["from_id"] could be None

File: backend/hct_mis_api/aurora/admin.py (hunk 3 above)
Comment (domdinicola) [reply backend/hct_mis_api/aurora/admin.py:220]: what would you Catch?

**Pull request closed without merging**